### ISSUE TITLE
S927 FP 

### ISSUE THREAD

**Issue by dmitry-golovinov-sonarsource** [SonarSource/sonar-dotnet#4370]
### Description

False positive for S927. When the type of the argument is generic interface it will be more useful to provide name based on the actual substituted type, since it can better express what the code is about

### Repro steps

```
    public interface SomeInterface<A> { }

    public interface BaseInterface<A>
    {
        void Apply(SomeInterface<A> param);
    }

    public class BasicImplementation : BaseInterface<int>
    {
        public void Apply(SomeInterface<int> intValue) { }   //Noncompliant FP
    }
```

### Expected behavior

Do not raise the rule.

### Actual behavior

In the previous example issue is being raised, since type kind of parameter **param** in **Apply** method of **BaseInterface**  
does not have the kind of **TypeParameter**.

### Related information

* SonarC# Version: 8.22
* Visual Studio 16.9.4
